wp-content/plugins/core/src/Templates/Factory_Method.php

```diff
@@ -0,0 +1,13 @@
+<?php
+declare( strict_types=1 );
+
+namespace Tribe\Project\Templates;
+
+trait Factory_Method {
+	/**
+	 * @return static
+	 */
+	public static function factory() {
```

Comment:
This is for the template to instantiate the controller, so it needs to be a static method we can call before we have an instance. And the method should reach out to the DI container to get the fully configured instance, instead of calling `new` in the template file.

Why a trait instead of a parent class? Because I rushed through this and haven't fully considered how factories are going to work. This is definitely part of the WIP.